### ISSUE TITLE
New update of VegaLite "broke" some functionality

### ISSUE THREAD

**Issue by davibarreira** [queryverse/VegaLite.jl#355]
I've the VegaLite.jl package, and a code that was working before is not working anymore. I went back to a fork that I had, and the fork is working, so indeed the problem is something related to the most recent commits to the package.

Here is an example of code that does not work anymore:
```
color_bar1= "black"
color_bar2 = "blue"
@vlplot(data=df,x=:x,y=:y,mark={type=:circle},color={field=:z,type="nominal",scale={range=[color_bar1,color_bar2]}}
```
This returns an error saying that the variable `color_bar1` is not declared. If I actually write the string instead of using the variable, the code runs.

**Comment by jlewis91** [comment]
@davibarreira Just wanted to update this issue with a reproducible example which both documents the problem and demonstrates an interim solution:

```julia
using VegaLite
using DataFrames

color_bar1= "black"
color_bar2 = "blue"
df = DataFrame(Dict("x"=> [1,2,3], "y"=> [2,3,4]))

# Works
range_list = ["black", "blue"]
@vlplot(data=df,x=:x,y=:y,mark={type=:circle},color={field=:z,type="nominal",scale={range=range_list}})

# Fails
color_bar1= "black"
@vlplot(data=df,x=:x,y=:y,mark={type=:circle},color={field=:z,type="nominal",scale={range=[color_bar1,"blue"]}})
```

**Comment by davibarreira** [comment]
Any ideas on why this is happening?

**Comment by jlewis91** [comment]
My guess: an issue with the environment scope for the variable, plus the variable being 'too many levels deep' in the dict / object.

**Comment by jlewis91** [comment]
@davibarreira I don't believe that this is an issue in the current version of the library, try updating?; will add a test for this sort of 'deep nesting' in a PR

**Comment by davibarreira** [comment]
Thanks. This is odd, cause I've updated and it's not working. And when I go back to the previous version it works.

**Comment by jlewis91** [comment]
You can see the test here, which passes the CI/CD: https://github.com/queryverse/VegaLite.jl/pull/359/files#diff-43b971f12d551dcec92d994a600e68a5ee8570e9e313e61713c8a3ca1b0587cdR107

@davidanthoff I think this can be closed?

**Comment by jlewis91** [comment]
@davibarreira Can you check and confirm the versions of Vega and vegalite you are using?

**Comment by davibarreira** [comment]
Right now I'm using a personal fork. I'll try updating back and see if everything works.

**Comment by davidanthoff** [comment]
So I'm not entirely clear what the status of this is :) Do we think that this is resolved?

**Comment by jlewis91** [comment]
IMHO this is resolved as soon as #359 is merged
